lib/core/types.hpp

```diff
@@ -1,7 +1,17 @@
 #pragma once
+#include <stdint.h>
 
 namespace hyped::core {
 
 enum class LowOrHigh { kLow = 0, kHigh };
 
+struct CanFrame
+{
+    uint32_t can_id; /* 32 bit CAN_ID + EFF/RTR/ERR flags */
+    uint8_t can_dlc; /* frame payload length in byte (0 .. CAN_MAX_DLEN) */
+    uint8_t __pad;   /* padding */
+    uint8_t __res0;  /* reserved / padding */
+    uint8_t __res1;  /* reserved / padding */
+    uint8_t data[8];
```

Comment:
`std::array<int8_t, 8>`?

Reply:
is that functionally different from data[8]? if it is, then will need to be tested to see if it works with socketCAN

Reply:
Not 100% sure. Just that `uint8_t data[8]` defines a C-style array but we're writing C++ so usually want to use C++ style arrays?

Reply:
i'll change it over, once we get socketCAN working then can test for sure

Reply:
std::array is just a wrapper for a generic c style array afaik so it shouldn't make a difference but yes testing it is a good idea